### PR TITLE
Refresh user information at start-up

### LINKED ISSUE #632: Fetching runtime data with existing login fails silently
Get `401 Unauthorized` from get runtime request. However this error is never surfaced to the user, but silently ignored. This can cause the runtime list to be outdated, and probably also fails on syncing projects.

Logging out and in a gain fixed the problem. So seems like a token expiration issue.

### PR DESCRIPTION
**Pull request by bergie**
Follow-up to #660. Will fix #632

Also adds a much clearer way for users to grant the app access to their GitHub repositories:

screenshot 2017-02-16 at 12 40 54

### CODE REVIEW

File: elements/noflo-account.html [hunk 1]
```diff
@@ -229,6 +272,31 @@
...
+        relogin: function (event, details, sender) {
+          event.preventDefault();
+          var scope = sender.getAttribute('data-scope');
+          this.fire('relogin', scope);
+        },
+        userChanged: function () {
+          if (!this.user) {
+            return;
+          }
+          if (!this.user.plan || this.user.plan.type === 'free') {
```
Comment (jonnor) [inline elements/noflo-account.html]: Why more logic in Polymer elements? We should be removing logic, not adding.

File: elements/noflo-account-settings.html [hunk 2]
```diff
@@ -74,90 +52,6 @@
       detached: function () {
         document.getElementById('container').classList.remove('blur');
       },
-      checkFlowhubUser: function (callback) {
```
Comment (jonnor) [inline elements/noflo-account-settings.html]: Good, less logic in view elements.

File: components/RefreshRemoteUser.coffee [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  , (data, groups, out, callback) ->
+    unless data?['grid-token']
+      # If user is not logged in, there is nothing to do
+      out.pass.send data
+      do callback
+      return
+
+    unless navigator.onLine
+      # When offline we can't refresh
+      out.pass.send data
```
Comment (jonnor) [inline components/RefreshRemoteUser.coffee]: I'd consider sending the not-logged-in and not-online cases to different ports. That way the semantics of the component is visible from outside, in the graph.

Comment (bergie) [inline components/RefreshRemoteUser.coffee]: This component doesn't really care whether logged in or not. The actual thing the component does should be pretty well reflected in the ports:

* User information was fine as-is (whether logged in or not)
* User information was invalid (non-working OAuth token etc)
* User information was updated from IdP

File: elements/noflo-account.html (hunk 1 above)
Comment (bergie) [inline elements/noflo-account.html]: This is UI-side logic so I found this to be the right place for it.

File: components/RefreshRemoteUser.coffee (hunk 3 above)
Comment (jonnor) [inline components/RefreshRemoteUser.coffee]: Ok

File: elements/noflo-account.html (hunk 1 above)
Comment (jonnor) [inline elements/noflo-account.html]: Storing `askForScope` in the widget introduces (more) statefulness. Is there at least a less stateful way to represent this transformation of user info to these scopes?
And is this the scopes we have asked for, or will ask for? The naming is not clear.

Comment (bergie) [inline elements/noflo-account.html]: These are scopes we offer to the user as options to allow. The "grant access to X" buttons are populated from this.

**Pull request merged**